### ISSUE TITLE
Option to mount data on NFS

### ISSUE THREAD

**Issue by cusspvz** [sameersbn/docker-gitlab#451]
Instead of hacking `entrypoing` and `command` it would be great to have options for mounting an NFS volume.

We will use it to make this compatible with [btsync image](https://github.com/cusspvz/btsync.docker)

**Comment by sameersbn** [comment]
you mean the image should support mounting  NFS volumes?
If yes, then IMO that is not the right solution. Instead you should create a data-only container which mounts the NFS volume and then use the volumes-from option to import the NFS mount point into the GitLab container.

**Comment by cusspvz** [comment]
If i had to choose between both, i would choose the `data` container and `volumes-from` options aswell.
Happens that it production and large scale things doesn't behave quite as we wanted them to. When loading gitlab as a standalone mode, data is there and you could access right away! If you're going to use that technique into a cluster, it doesn't quite work because you have to sync all data before running up `GitLab`, otherwise it will try to compile assets and create/migrate database. As so, last option is to access to a network file system, where data scaling isn't directly related with GitLab, and thats where NFS enters with BTSync.

Changed my proposal to allow any type of data mounting before initializing GitLab. This allow users to select between [BTSync + NFS](https://github.com/cusspvz/btsync.docker) or any other implementation such as GlusterFS.

Docker has already [some options for volume drivers using NFS](https://github.com/SvenDowideit/docker-volumes-nfs), but it isn't ready for production.

**Comment by frank-dspeed** [comment]
NO YOUR ALL WRONG Look this:
https://news.ycombinator.com/item?id=9343021

gitlab will use git git will support alternate storage like couchbase or amazon s3 so gitlab itself will and then there is no mount workaround needed.

**Comment by frank-dspeed** [comment]
you can close this or move this to the existing PR's at git and gitlab but we are implamenting it even because we know the issu so only for your education current implamentation of gitlab and git supports alternate storage engines then a posix file system ;)

**Comment by cusspvz** [comment]
@frank-dspeed could you please point to the exact place where it is being discussed? For me, it seems they are only discussing GitHub LFS (storage for none-code resources that are usually large, such as DBs, images, etc) and relating it with older software.

**Comment by sameersbn** [comment]
i agree, the discussion in the ycombinator link is something completely different. does not apply here.

@cusspvz i dont completely understand your use case. Having all gitlab data on a centralized NFS server and mounting it as a volume does not work? What is the use of BTSync here.

**Comment by cusspvz** [comment]
We're deploying into a cluster, centralizing something isn't an option for us. As so, I've created a image that sync files cross containers, and called btsync. Problem is that we were getting lot of issues trying to deploy as a "Global" cluster service. As so, we added the NFS server feature to allow other containers to access BTSync containers as a NFS service, being a network resource like mysql, redis, or any other else.

Since this image doesn't have NFS support, we are sidekicking an image to mount NFS and expose as a Volume, but thats a bad workaround as hell.

**Comment by frank-dspeed** [comment]
http://blog.deveo.com/your-git-repository-in-a-database-pluggable-backends-in-libgit2/

**Comment by frank-dspeed** [comment]
Sorry for the late answer but i have realy less time but i think there you can read up a bit whats it all about you can store all not only the meta data in databases we at present work mainly on couchbase because its the successor over simply all database systems on earth including all apache big data projects

expect auto clustering, sql like quering via n1ql only to point to some key futures, sure it supports also persistence as in memory storage read it up its worth the reading.

we ported all our storage needs to our own s3 implamentation via couchbase and cbfs we also doing a fuse mount for couchbase file system (cbfs)  but using libgit2 and directly use couchbase is the best.
